lib/services/locator/locator_service_register.dart

```diff
@@ -3,5 +3,6 @@ part of 'locator_service.dart';
 class LocatorServiceRegister {
   LocatorServiceRegister() {
     locator.registerFactory<HttpService>(() => HttpServiceImpl());
+    locator.registerFactory<ApiService>(() => ApiServiceImpl());
```

Comment:
again, I feel like you're doing quite a lot of legwork here instead of using HTTP client libs. But if you want to go this way for the IC learning I'm ok with that.

Reply:
@sleepylee Which `HTTP client libs` that you mention here? I will take a look at it to have a comparison 😁